extendr-api/src/robj/into_robj.rs

```diff
@@ -33,16 +33,63 @@ impl From<()> for Robj {
 ///     assert_eq!(r!(my_func()), r!(1.0));
 /// }
 /// ```
-impl<T> From<Result<T>> for Robj
+// impl<T> From<Result<T>> for Robj
+// where
+//     T: Into<Robj>,
+// {
+//     fn from(res: Result<T>) -> Self {
+//         match res {
+//             Ok(x) => x.into(),
+//             Err(x) => x
+//                 .to_string()
+//                 .into_robj()
+//                 .set_attrib("extendr_err", true)
+//                 .unwrap(),
+//         }
+//     }
+// }
+
+// it does not have to be extendr result
+
+impl<T, E> From<std::result::Result<T, E>> for Robj
 where
     T: Into<Robj>,
+    E: Into<Robj>,
 {
-    fn from(res: Result<T>) -> Self {
-        // Force a panic on error.
-        res.unwrap().into()
+    fn from(res: std::result::Result<T, E>) -> Self {
+        match res {
+            Ok(x) => x.into(),
+            Err(x) => x.into().set_attrib("extendr_err", true).unwrap(),
```

Comment:
I actually think we should handle the attribute setting error too, I know it will be a tad annoying to implement.

Reply:
If some Robj actually would fail on set_attrib() instead of panic! this code could another attribute error stating an "internal error". Maybe it is acceptable to panic on internal errors. I will try to look into when set_attrib could fail.

Reply:
I have tried to address set_attrib. I have been thinking about handling errors from set_attrib() list!() set_class(). If e.g. not possible to set attribute on an error or class on a condtion, then it is possible to make a new simpler Robj describing this as an internal error. However, if it is still not possible to create attributes or conditions (e.g. due to mem alloc), there is nothing left to do than panicking.

Reply:
https://github.com/extendr/extendr/blob/aecff2890b4090c70fa80c54d870716d8f1e0537/extendr-api/src/robj/into_robj.rs#L91

Reply:
Follow-up: 
If using feature `result_condition` then the error is tagged as @multimeric  suggested 
https://github.com/sorhawell/extendr/blob/4a7a4dd623a0446ee2ef2d8c6684f6b006dbc323/extendr-api/src/robj/into_robj.rs#L103

if using feature `result_list` then the `list(ok=?,err=?)` is tagged with class `extendr_result`
https://github.com/sorhawell/extendr/blob/4a7a4dd623a0446ee2ef2d8c6684f6b006dbc323/extendr-api/src/robj/into_robj.rs#L148

... and default with neither feature will just panic! on Err